wordvecspace/annoy.py

```diff
@@ -85,23 +89,196 @@ def get_distances(self, row_words_or_indices: Union[int, str, np.ndarray],\
+        v_w_i -- vector(s), word(s), index(es)
+        k -- number of nearest items to be retrieved
+        examine_k -- number of items to be examined only in set_intersect and set_union
+        combination -- combination to be done or not
+        combination_method -- set_intersect/set_union/distance/vector
+        weights -- importance factor
+        metric -- angular/cosine/euclidean
+        include_distances -- know the distance value or not
+
+        Tests:
+        >>> wv = WordVecSpaceAnnoy(DATAFILE_ENV_VAR)
+        >>> wv.get_nearest('india', k=5)
+        array([  509,   486,  6186, 22151,  1980], dtype=uint32)
+        >>> wv.get_nearest(709, k=5)
+        array([  709, 15824, 26752,  7597,   242], dtype=uint32)
+        >>> vecs = wv.get_vectors(['india', 'pakistan'])
+        >>> vecs
+        array([[-2.4544, -0.1309,  8.9653, -3.1779,  3.2016],
+               [-3.3736,  0.4845,  9.7016, -3.5337,  1.0142]], dtype=float32)
+        >>> wv.get_nearest(vecs, k=5)
+        array([[  509,   486,  6186, 22151,  1980],
+               [ 2224,  5281, 11087, 16342, 38760]], dtype=uint32)
+        >>> wv.get_nearest(vecs, k=5, examine_k=500,
+        ... combination=True, combination_method='set_intersect')
+        array([[ 2224,   523,  5969, 24149,  9772]])
+        >>> wv.get_nearest(vecs, k=5, examine_k=500,
+        ... combination=True, combination_method='set_intersect', weights=[0.1, 0.9])
+        array([[ 2224,  5281,  3560,  4622, 11886]])
+        >>> wv.get_nearest(vecs, k=5, examine_k=500, combination=True,
+        ... combination_method='set_intersect', weights=[0.1, 0.9], include_distances=True)[1]
+        array([[0.024 , 0.0844, 0.1039, 0.1054, 0.1067]], dtype=float32)
+        >>> wv.get_nearest(vecs, k=5, examine_k=500,
+        ... combination=True, combination_method='set_union')
+        array([[  509,   486,  6186, 22151,  1980]], dtype=uint32)
+        >>> wv.get_nearest(vecs, k=5, examine_k=500,
+        ... combination=True, combination_method='set_union', weights=[0.9, -0.9])
+        array([[64001, 12629, 65521, 56911,  9621]], dtype=uint32)
+        >>> wv.get_nearest(vecs, k=5, examine_k=500, combination=True,
+        ... combination_method='set_union', weights=[0.1, 0.9], include_distances=True)[1]
+        array([[0.    , 0.0058, 0.0077, 0.01  , 0.011 ]], dtype=float32)
+        >>> wv.get_nearest(vecs, k=10, combination=True, combination_method='vector')
+        array([  523,  5969,  4622, 18072,  2224, 10018,  5226,  8012,  5281,
+               17910], dtype=uint32)
+        >>> wv.get_nearest(vecs, k=10, combination=True,
+        ... combination_method='vector', include_distances=True)[1]
+        array([0.0066, 0.0658, 0.1132, 0.1134, 0.1154, 0.1315, 0.1343, 0.1544,
+               0.1579, 0.169 ], dtype=float32)
+        """
+
+        if not combination:
+            return self._get_brute(v_w_i, k, include_distances)
+
+        if combination and combination_method == 'set_intersect':
```

Comment:
I think we don't need to check if `combination` is there or not here
Because if `combination` is not there it won't come here.
Same goes for below two conditions also